src/main/kotlin/com/faforever/icebreaker/service/SessionService.kt

```
@@ -27,25 +28,34 @@ class SessionService(
     @Transactional
     fun getSession(gameId: Long): Session {
         try {
-            iceSessionRepository.acquireGameLock(gameId)
+            when (iceSessionRepository.acquireGameLock(gameId)) {
+                true -> {
+                    val session = iceSessionRepository.findByGameId(gameId)
+                        ?: IceSessionEntity(gameId = gameId, createdAt = Instant.now()).also {
+                            LOG.debug("Creating session for gameId $gameId")
+                            iceSessionRepository.persist(it)
+                        }
 
-            val session = iceSessionRepository.findByGameId(gameId)
-                ?: IceSessionEntity(gameId = gameId, createdAt = Instant.now()).also {
-                    LOG.debug("Creating session for gameId $gameId")
-                    iceSessionRepository.persist(it)
-                }
+                    val servers = activeSessionHandlers.flatMap {
+                        it.createSession(session.id)
+                        it.getIceServersSession(session.id)
+                    }
 
-            val servers = activeSessionHandlers.flatMap {
-                it.createSession(session.id)
-                it.getIceServersSession(session.id)
+                    return Session(
+                        id = session.id,
+                        servers = servers,
+                    )
+                }
+                else -> {
+                    throw TimeoutException("Unable to acquire game lock for $gameId")
```

If you want to throw an exception, might as well do it in IceSessionRepository